ext/strscan/strscan.c

```diff
@@ -1663,6 +1663,9 @@ strscan_matched_size(VALUE self)
 static int
 name_to_backref_number(struct re_registers *regs, VALUE regexp, const char* name, const char* name_end, rb_encoding *enc)
 {
+    if (!RTEST(regexp))
+        rb_enc_raise(enc, rb_eIndexError, "undefined group name reference: %.*s",
+                     rb_long2int(name_end - name), name);
     int num;
 
     num = onig_name_to_backref_number(RREGEXP_PTR(regexp),
```

Comment:
Can we reuse existing `rb_enc_raise()`?

```suggestion
    int num = -1;

    if (RTEST(regexp) {
      num = onig_name_to_backref_number(RREGEXP_PTR(regexp),
```

Reply:
It's nice!